source/Coop/Mod/DebugUtil/MoveTo.cs

```diff
@@ -0,0 +1,283 @@
+            if (moveTo != null)
+            {
+                if (CoopServer.Instance.Persistence.Room.Entities.
+                    FirstOrDefault(q => q.Id == moveTo.entityId) is MobilePartyEntityServer entity)
+                    entity.State.Movement = new MovementState()
+                    {
+                        DefaultBehavior = moveTo.movement.DefaultBehaviour,
+                        Position = moveTo.movement.TargetPosition,
+                        SettlementIndex = moveTo.movement.TargetSettlement != null ?
+                            moveTo.movement.TargetSettlement.Id : MovementState.InvalidIndex,
+                        TargetPartyIndex = moveTo.movement.TargetParty != null ?
+                            moveTo.movement.TargetParty.Id : MovementState.InvalidIndex
+                    };
+
+                moveTo.passed = true;
+                Logger.Warn("[MOVETO] Moving to new position.");
+            }
+
+            if (now >= lastTick)
+            {
+                Stop(); // TODO: send message in debug console instead to send on game screen
+                Logger.Warn("[MOVETO] Playback has finished.");
+            }
+        }
+
+        private static void VerifyEvents()
+        {
+            var difftime = new List<long?>();
+
+            foreach (var play in PlaybackEventList)
+            {
+                var rec = RecordingEventList.FirstOrDefault(q => !q.passed && q.Equals(play));
+                if (rec != null)
+                {
+                    difftime.Add((long)(rec.time - play.time).ToMilliseconds);
+                    rec.passed = true;
+                }
+                else
+                    difftime.Add(null);
+            }
+
+            var csv = difftime.Where(q => q != null).Select(q => new object[6] { q, 0, null, null, null, null }).ToList();
+            var skipped = difftime.Count(q => q == null);
+            var new_events = RecordingEventList.Count(q => !q.passed);
+
+            int l_minValue = (int)difftime.Min().Value;
+            int l_maxValue = (int)difftime.Max().Value;
+
+            int minValue = Math.Sign(l_minValue) * RailUtil.Log2((ulong)Math.Abs(l_minValue));
+            int maxValue = Math.Sign(l_maxValue) * RailUtil.Log2((ulong)Math.Abs(l_maxValue)) + 1;
+
+            int colsCount = csv.Count / 5;
+            int colWidth = (maxValue - minValue) / colsCount;
+
+            csv[0][2] = skipped;
+            csv[0][3] = new_events;
+            csv[0][4] = l_minValue;
+            csv[0][5] = l_maxValue;
+
+            foreach (var line in csv)
+            {
+                long diff = ((long?)line[0]).Value;
+                int index = (Math.Sign(diff) * RailUtil.Log2((ulong)Math.Abs(diff)) - minValue) / colWidth;
+                csv[index][1] = (int)csv[index][1] + 1;
+            }
+
+            var report = $"time_diff(ms),cols,skipped,new_events,minValue,maxValue\r\n";
```

Comment:
The `time_diff(ms)` is in campaign time. Can we maybe get a real world time measurements for this? i think it would be much easier to undestand the results since 1ms in game is basically nothing in real time.

Reply:
@araex Campaign time is more matter for right simulating events. The main thing is that the variance of the time intervals was to be minimal.